SignalModelInterpolation/create_signal_ws_new_cat_2d_res_bkg.py

```diff
@@ -160,8 +159,10 @@ def main(args):
       cats = sorted(models[proc][year].keys())
       for cat in cats:
         out_path = os.path.join(args.outdir, "%s_%s_cat%s.root"%(proc, year, cat))
-#        makeWorkspace(models[proc], systematics[proc], year, cat, out_path, args.mgg_range, proc, args.doSyst)
-        makeWorkspace(models[proc], systematics, year, cat, out_path, args.mgg_range, proc, args.doSyst)
+        if args.doSyst:
+            makeWorkspace(models[proc], systematics[proc], year, cat, out_path, args.mgg_range, proc, args.doSyst)
+        else:
+            makeWorkspace(models[proc], systematics, year, cat, out_path, args.mgg_range, proc, args.doSyst)
```

Comment:
These lines are confusing. What's the point of changing the systematics argument? What's the point of passing the `args.doSyst` in the function if you are to run a different version with an `if` clause?

Maybe try:
```suggestion
        makeWorkspace(models[proc], systematics, year, cat, out_path, args.mgg_range, proc, args.doSyst)
```
and then add the following line at the beginning of the `makeWorkspace` function:
```
systematicss = None if not doSyst
```
and see if it works?

Reply:
I wrote a new way in the code. The issue is that if we do not do systematics, it will try to fetch an empty array to get systematics[proc], which will give crash. I define a new array systematics_proc, if doSyst, it is given systematics[proc], otherwise it is given None.